server/controllers/credit_card_controller.py

```diff
@@ -14,6 +14,15 @@ def get_credit_card_by_number(self, card_number):
         credit_card = CreditCard.query.filter_by(card_number=card_number).first()
         return jsonify(credit_card.to_dict() if credit_card else {})
 
+    def get_credit_card_balance(self, card_number):
+        credit_card = CreditCard.query.filter_by(card_number=card_number).first()
+        if credit_card:
+            # Sum up the amounts of all transactions related to the credit card
+            total_balance = sum(transaction.amount for transaction in credit_card.transactions)
```

Comment:
Style thing: For a best practice point of view, would it make ssense that a credit card can sum up its balance? And we simply call that method here instead of doingn logic in controller.
 
cc senior dev / staff